tests/Finder/LscpuPhysicalFinderTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Fidry CPUCounter Config package.
+ *
+ * (c) Théo FIDRY <theo.fidry@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Fidry\CpuCoreCounter\Test\Finder;
+
+use Fidry\CpuCoreCounter\Finder\CpuCoreFinder;
+use Fidry\CpuCoreCounter\Finder\LscpuPhysicalFinder;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Fidry\CpuCoreCounter\Finder\LscpuLogicalFinder
+ *
+ * @internal
+ */
+final class LscpuPhysicalFinderTest extends TestCase
+{
+    /**
+     * @dataProvider finderProvider
+     */
+    public function test_it_can_describe_itself(CpuCoreFinder $finder, string $expected): void
+    {
+        $actual = $finder->toString();
+
+        self::assertSame($expected, $actual);
+    }
+
+    public static function finderProvider(): iterable
+    {
+        yield [
+            new LscpuPhysicalFinder(),
+            FinderShortClassName::get(new LscpuPhysicalFinder())
+        ];
+    }
+
+    /**
+     * @dataProvider lscpuProvider
+     */
+    public function test_it_can_count_the_number_of_cpu_cores(
+        string $lscpu,
+        ?int $expected
+    ): void {
+        $actual = LscpuPhysicalFinder::countCpuCores($lscpu);
+
+        self::assertSame($expected, $actual);
+    }
+
+    public static function lscpuProvider(): iterable
+    {
+        yield 'example with four logical but two physical' => [
+            <<<'EOF'
+# The following is the parsable format, which can be fed to other
+# programs. Each different item in every column has an unique ID
+# starting from zero.
+# CPU,Core,Socket,Node,,L1d,L1i,L2,L3
+0,0,0,0,,0,0,0,0
+1,1,0,0,,1,1,1,0
+2,0,0,0,,0,0,0,0
+3,1,0,0,,1,1,1,0
+
+EOF
+            ,
+            2
+        ];
+
+        yield 'example with two cores' => [
+            <<<'EOF'
+# The following is the parsable format, which can be fed to other
+# programs. Each different item in every column has an unique ID
+# starting from zero.
+# CPU,Core,Socket,Node,,L1d,L1i,L2
+0,0,0,0,,0,0,0
+1,1,0,0,,1,1,1
+
+EOF
+            ,
+            2
+        ];
+
+
+        yield 'example with unrecognized physical core' => [
+            <<<'EOF'
+# The following is the parsable format, which can be fed to other
+# programs. Each different item in every column has an unique ID
+# starting from zero.
+# CPU,Core,Socket,Node,,L1d,L1i,L2
+0,0,0,0,,0,0,0
+1,-,0,0,,1,1,1
```

Comment:
This is tricky. I am not certain it comes out exactly like this, neither that it means its only one physical core. I am certain lscpu might output `-` on some cases as I can understand from the source code.

Reply:
checking on docker `ubuntu:rolling`, I get that there:

```
# CPU,Core,Socket,Node
0,0,0,
1,1,0,
2,2,0,
```

So yes I think the output can vary a bit

Reply:
Yes, the 1st column will be the logical core and the 2nd column will be the physical so it might be the case of something like:

```
# CPU,Core,Socket,Node
0,0,0,
1,1,0,
2,0,0,
3,1,0,
```

But when for some reason it is not recognized it will output `-`. See line https://github.com/util-linux/util-linux/blame/master/sys-utils/lscpu.c#L303 in the latest `lscpu` that exists for some years. Given that what I say now is correct, currently the code handles this as "unrecognized" and does not count it. But I am not sure how I can get certain 100%. I tried a couple of things and searched the internet for this output but couldn't find it. I suppose I am correct as the man page shows the same

```
-y, --physical
           Display physical IDs for all columns with topology elements (core, socket, etc.). Other than logical IDs, which are assigned by lscpu, physical
           IDs are platform-specific values that are provided by the kernel. Physical IDs are not necessarily unique and they might not be arranged
           sequentially. If the kernel could not retrieve a physical ID for an element lscpu prints the dash (-) character.
```